src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -461,7 +465,44 @@ async def test_TC_TSTAT_4_2(self):
             # Roll back
             await self.send_atomic_request_rollback_command()
 
-        # TODO: Add tests for the total number of Presets exceeds the NumberOfPresets supported. Also Add tests for adding presets with preset scenario not present in PresetTypes.
+        self.step("17")
+        if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
+            test_presets = new_presets_with_handle.copy()
+            test_presets.append(cluster.Structs.PresetStruct(presetHandle=NullValue, presetScenario=9,
+                                                             name="Wake", coolingSetpoint=2500, heatingSetpoint=1700, builtIn=False))
+
+            # Send the AtomicRequest begin command
+            await self.send_atomic_request_begin_command()
+
+            await self.write_presets(endpoint=endpoint, presets=test_presets, expected_status=Status.ConstraintError)
+
+            # Clear state for next test.
+            await self.send_atomic_request_rollback_command()
+
+        self.step("18")
+        if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
+
+            # Read the active preset handle attribute and verify it was updated to preset handle
```

Comment:
That is ... not the attribute that is being read?

Reply:
fixed.